src/OrleansRuntime/Core/InsideGrainClient.cs

```diff
@@ -750,5 +750,16 @@ public SiloStatus GetSiloStatus(SiloAddress siloAddress)
         {
             return Silo.CurrentSilo.LocalSiloStatusOracle.GetApproximateSiloStatus(siloAddress);
         }
+
+        public void BreakOutstandingMessagesToDeadSilo(SiloAddress deadSilo)
+        {
+            foreach (var callback in callbacks)
+            {
+                if (callback.Value.Message.TargetSilo.Equals(deadSilo))
```

Comment:
just to be 100% safe, check that `callback.Value.Message.TargetSilo` is not null (instead of accessing twice, store in local var first, since `Value.Message` is a dictionary lookup)